ios/RNTrackPlayer/Models/SessionCategories.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  SessionCategory.swift
+//  RNTrackPlayer
+//
+//  Created by Thomas Hessler on 3/12/19.
+//  Copyright © 2019 David Chavez. All rights reserved.
+//
+
+import Foundation
+import MediaPlayer
+import AVFoundation
+
+enum SessionCategory : String {
+    
+    case playAndRecord, multiRoute, playback, ambient, soloAmbient
+    
+    func mapConfigToAVAudioSessionCategory() -> AVAudioSession.Category {
+        switch self {
+        case .playAndRecord:
+            return .playAndRecord
+        case .multiRoute:
+            return .multiRoute
+        case .playback:
+            return .playback
+        case .ambient:
+            return .ambient
+        case .soloAmbient:
+            return .soloAmbient
+        }
+    }
+}
+
+enum SessionCategoryOptions : String {
+    
+    case mixWithOthers, duckOthers, interruptSpokenAudioAndMixWithOthers, allowBluetooth, allowBluetoothA2DP, allowAirPlay, defaultToSpeaker
+
+    func mapConfigToAVAudioSessionCategoryOptions() -> AVAudioSession.CategoryOptions? {
+        switch self {
+        case .mixWithOthers:
+            return .mixWithOthers
+        case .duckOthers:
+            return .duckOthers
+        case .interruptSpokenAudioAndMixWithOthers:
+            return .interruptSpokenAudioAndMixWithOthers
+        case .allowBluetooth:
+            return .allowBluetooth
+        case .allowBluetoothA2DP:
+            return .allowBluetoothA2DP
+        case .allowAirPlay:
+            return .allowAirPlay
+        case .defaultToSpeaker:
+            return .defaultToSpeaker
+        }
+    }
+}
+
+enum SessionCategoryMode : String {
+    
+    case `default`, gameChat, measurement, moviePlayback, spokenAudio, videoChat, videoRecording, voiceChat, voicePrompt
+    
+    func mapConfigToAVAudioSessionCategoryMode() -> AVAudioSession.Mode {
+        switch self {
+        case .default:
+            return .default
+        case .gameChat:
+            return .gameChat
+        case .measurement:
+            return .measurement
+        case .moviePlayback:
+            return .moviePlayback
+        case .spokenAudio:
+            return .spokenAudio
+        case .videoChat:
+            return .videoChat
+        case .videoRecording:
+            return .videoRecording
+        case .voiceChat:
+            return .voiceChat
+        case .voicePrompt:
+            if #available(iOS 12.0, *) {
+                return .voicePrompt
+            } else {
```

Comment:
Is this else needed then? 😉 Let's just omit it and treat it as invalid options for `< iOS 12` the logic in the setup method can handle this just fine.

Reply:
my XCode was throwing an error here because it's expecting a return. that's why I was defaulting to `.default`. Is there a way to catch for it differently?

Reply:
ah, interesting.. there probably is, but I can't think of it at the moment, so let's leave it as-is 👍